src/ert/config/observations.py

```diff
@@ -1,471 +1,241 @@
-    import numpy.typing as npt
-
-    from .ensemble_config import EnsembleConfig
-
-DEFAULT_TIME_DELTA = timedelta(seconds=30)
+from .response_properties import ResponseTypes
 
 
 def history_key(key: str) -> str:
     keyword, *rest = key.split(":")
     return ":".join([keyword + "H"] + rest)
 
 
+class _AccumulatedDataset(Protocol):
+    def to_xarray(self) -> xr.Dataset: ...
+    def __len__(self) -> int: ...
+
+
+@dataclass
+class _SummaryObsDataset:
+    summary_keys: List[str] = field(default_factory=list)
+    observations: List[float] = field(default_factory=list)
+    stds: List[float] = field(default_factory=list)
+    times: List[Union[int, datetime]] = field(default_factory=list)
+    obs_names: List[str] = field(default_factory=list)
+
+    def __len__(self) -> int:
+        return len(self.summary_keys)
+
+    def to_xarray(self) -> xr.Dataset:
+        return (
+            pd.DataFrame(
+                data={
+                    "name": self.summary_keys,
+                    "obs_name": self.obs_names,
+                    "time": self.times,
+                    "observations": self.observations,
+                    "std": self.stds,
+                },
+            )
+            .set_index(["name", "obs_name", "time"])
+            .to_xarray()
+        )
+
+
+@dataclass
+class _GenObsDataset:
+    gen_data_keys: List[str] = field(default_factory=list)
+    observations: List[float] = field(default_factory=list)
+    stds: List[float] = field(default_factory=list)
+    indexes: List[int] = field(default_factory=list)
+    report_steps: List[int] = field(default_factory=list)
+    obs_names: List[str] = field(default_factory=list)
+
+    def __len__(self) -> int:
+        return len(self.gen_data_keys)
+
+    def to_xarray(self) -> xr.Dataset:
+        return (
+            pd.DataFrame(
+                data={
+                    "name": self.gen_data_keys,
+                    "obs_name": self.obs_names,
+                    "report_step": self.report_steps,
+                    "index": self.indexes,
+                    "observations": self.observations,
+                    "std": self.stds,
+                }
+            )
+            .set_index(["name", "obs_name", "report_step", "index"])
+            .to_xarray()
+        )
+
+
+class _GenObsAccumulator:
+    def __init__(self) -> None:
+        self.ds: _GenObsDataset = _GenObsDataset()
+
+    def write(
+        self,
+        gen_data_key: str,
+        obs_name: str,
+        report_step: int,
+        observations: List[float],
+        stds: List[float],
+        indexes: List[int],
+    ) -> None:
+        self.ds.gen_data_keys.extend([str(gen_data_key)] * len(observations))
+        self.ds.obs_names.extend([str(obs_name)] * len(observations))
+        self.ds.report_steps.extend([report_step] * len(observations))
+
+        self.ds.observations.extend(observations)
+        self.ds.stds.extend(stds)
+        self.ds.indexes.extend(indexes)
```

Comment:
Suggest just moving this to: `_GenObsDataset`, dont quite see the need for an accumulator class.